### ISSUE TITLE
AI gets easily protect/switch baited

### ISSUE THREAD

**Issue by DizzyEggg** [rh-hideout/pokeemerald-expansion#3346]
### Description

I knew our AI was easy to exploit, but this is ridiculously bad. AI is too predictable with its move selection when switching is involved. Here's a gif which illustrates the issue perfectly.

![Ai protect switch bait](https://github.com/rh-hideout/pokeemerald-expansion/assets/16259973/25739c3e-894a-4f4e-b4c5-07b75e2641e2)

I have some ideas on how to fix it, but I'm open to hear what you guys have in mind. I also plan to have AI tests in the near future, so we would have more control over different AI scenarios, without the need to test everything by hand.

### Version

1.5.3 (Default)

**Comment by AlexOn1ine** [comment]
As far as I know most hacks that have some form of anti abuse AI work with counters (that are increased on switching/no damage takes). Once it reaches a certain number, the player will be punished for it.

**Comment by DizzyEggg** [comment]
> As far as I know most hacks that have some form of anti abuse AI work with counters (that are increased on switching/no damage taken). Once it reaches a certain number, the player will be punished for it.

Yeah, counters are a good idea. There's already a switch-in counter which I added back in(though it's never used at this point) #2230. 
As for punishing it depends what you mean by it. Ideally I'd want AI to 1) predict this may happen and 2) react appropriately. 
Maybe something like using a move which hits hard the predicted switch mon. Or AI would themselves switch to a different mon, or use Roar. Obviously, it would need to be random, so the player couldn't predict what AI does.

**Comment by ShadowzLmao2** [comment]
I think that being able to abuse the AI is key to making good strategies in difficult ROMHacks. An AI like Radical Red where it punishes you after enough switches as Alex mentions is a good idea. One change to RR's AI is that protect turns shouldnt count as turns between switches. Personally, in the scenario above, I would EQ the stunfisk and tbolt the pelipper but after player switches back into stunfisk, i would then make a safe play and pick the move that does the most to stunfisk while hitting pelipper and worse case there is no move that hits both, go for the hard predict into pelipper. It would have to be done in a realistic fashion and not based on % chances to do something otherwise playing a ROMHack would no longer be fun and only luck dependant. if the ai is manipulatable, it makes good videos

**Comment by AlexOn1ine** [comment]
> I think that being able to abuse the AI is key to making good strategies in difficult ROMHacks. An AI like Radical Red where it punishes you after enough switches as Alex mentions is a good idea. One change to RR's AI is that protect turns shouldnt count as turns between switches. 

I don't think we should port the CFRU (RR) behavior at all and come up with our own ideas.

**Comment by hedara90** [comment]
This might've been fixed, or at least mitigated.
The AI no longer gets completely manipulated when switching around.
![switches](https://github.com/user-attachments/assets/78b95075-6d8d-4f5c-b463-4321039fc209)

**Comment by pkmnsnfrn** [comment]
In the [recent poll](https://discord.com/channels/419213663107416084/1266027326882185248), this specific issued was mentioned twice, once by Pawkkie and once by Awakeon

**Comment by Pawkkie** [comment]
Closed by #6743, hedara is the ![Image](https://github.com/user-attachments/assets/a6d81829-c9a1-4e58-8ddf-6cafeaddfddc)
